converter/shieldhit/parser.py

```diff
@@ -26,6 +26,8 @@ def save_configs(self, target_dir: str):
         Save the configs as text files in the target_dir.
         The files are: beam.dat, mat.dat, detect.dat and geo.dat.
         """
+        if not path.exists(target_dir):
```

Comment:
As for checking if target dir exists, I would use the following snippet, preferring `pathlib` over old-fashioned `os.path`:
```
from pathlib import Path
(...)
if not Path(target_dir).exists():
   (...)
```
see https://docs.python.org/3/library/pathlib.html

On the other hand, do we really need to be that restrictive?
I would simply create that directory if not existing and not raise the exception

With `pathlib` the whole story can be done in a single line, for inspiration see:
https://stackoverflow.com/questions/273192/how-can-i-safely-create-a-nested-directory


```
Path(target_dir).mkdir(parents=True, exist_ok=True)
```
for the description of the arguemtns, see https://docs.python.org/3/library/pathlib.html#pathlib.Path.mkdir